src/codeu/chat/server/Controller.java

```diff
@@ -151,25 +148,24 @@ private Uuid createId() {
 
     Uuid candidate;
 
-    for (candidate = uuidGenerator.make();
-         isIdInUse(candidate);
-         candidate = uuidGenerator.make()) {
+    for (candidate = uuidGenerator.make(); isIdInUse(candidate); candidate = uuidGenerator.make()) {
```

Comment:
I remember coming across this and thinking that it looked weird. To be honest, I wasn't even quite sure what it did. After rewriting it with a do/while, it makes much more sense though!